internal/v6provider/resource_load_balancer.go

```diff
@@ -136,6 +184,35 @@ func (r LoadBalancerResource) Schema(
 			"https_redirect": schema.BoolAttribute{
 				Computed: true,
 			},
+			"external_rules": schema.BoolAttribute{
+				Optional: true,
+				Description: "When enabled, The full list of rules are not " +
+					"managed by Terraform. Induvidual rules can still be " +
+					"managed with the katapult_load_balancer_rule " +
+					"resource. This is required to prevent Terraform from " +
+					"deleting rules managed outside of Terraform. Defaults " +
+					"to false.",
+				MarkdownDescription: "When enabled, " +
+					"The full list of rules are not " +
+					"managed by Terraform. Induvidual rules can still be " +
+					"managed with the `katapult_load_balancer_rule` " +
+					"resource. This is required to prevent Terraform from " +
+					"deleting rules managed outside of Terraform. Defaults " +
+					"to `false`.",
+				Validators: []validator.Bool{
+					boolvalidator.ConflictsWith(path.MatchRoot("rules")),
+				},
+			},
+			"rules": schema.ListNestedAttribute{
+				Optional: true,
+				Validators: []validator.List{
+					listvalidator.ConflictsWith(
+						path.MatchRoot("external_rules")),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: lbrSchema,
+				},
+			},
```

Comment:
Did we discuss about using nested objects?

https://developer.hashicorp.com/terraform/plugin/framework/handling-data/blocks/list-nested

I think we should use blocks for rules, so we have a `rule` block that can be repeated multiple times, similar to the `inbound_rule` and `outbound_rule` blocks in Security Groups.

We should most likely do the same for `virtual_machine`, `virtual_machine_group`, and `tag` too.

Reply:
Right, we had discussed this, and blocks are not recommended for use anymore in the Plugin Framework.